src/braket/ahs/atom_arrangement.py

```diff
@@ -72,3 +73,23 @@ def __iter__(self) -> Iterator:
 
     def __len__(self):
         return self._sites.__len__()
+
+    def discretize(
+        self,
+        position_res: Decimal
+    ) -> AtomArrangement:
+        """Creates a discretized version of the atom arrangement,
+        rounding all site coordinates to the closest multiple of the
+        resolution. The types of the sites are unchanged.
+
+        Args:
+            position_res (Decimal): Position resolution
+
+        Returns:
+            AtomArrangement: A new discretized atom arrangement.
+        """
+        discretized_arrangement = AtomArrangement()
+        for site in self._sites:
+            new_coordinates = tuple((round(Decimal(c) / position_res) * position_res for c in site.coordinate))
```

Comment:
Will the `new_coordinates` have the same type as the `site.coordinate`, and would it be `Decimal` or just `float`? I have the same question for the discretizers below, but if this one is fine, then the rest should be fine.

Reply:
1. Since both `Decimal(c)` and `position_res` are of `Decimal` type, the arithmetic operations with them results in a `Decimal`.
2. Yes, the site_type will stay the same (See line 94)